judge/judgedaemon.main.php

```diff
@@ -46,6 +46,28 @@ function read_credentials()
     }
 }
 
+
+function setup_curl_handle()
+{
+    global $curl_handle, $restuser, $restpass;
```

Comment:
one curl_handle is not enough. We need one per endpoint, right?

Reply:
yes, could create one per endpoint, or move the basic auth credential setting out of this function